frontend/src/features/payment/stripe/utils.ts

```diff
@@ -0,0 +1,10 @@
+import { PaymentIntent } from '@stripe/stripe-js'
+
+export const getPaymentViewType = (
+  status: PaymentIntent.Status | undefined,
+) => {
+  if (!status) return 'invalid'
+  if (['succeeded'].includes(status)) return 'receipt'
+  if (['canceled'].includes(status)) return 'canceled'
+  return 'payment'
+}
```

Comment:
wondering if these states should be extracted to an enum

Reply:
9bba127